untitled/src/com/teachmeskills/service/productService/ShoppingCartService.java

```diff
@@ -0,0 +1,146 @@
+package com.teachmeskills.service.productService;
+
+import com.teachmeskills.entity.shop.ShoppingCart;
+import com.teachmeskills.entity.products.Product;
+import com.teachmeskills.entity.shop.Shop;
+import com.teachmeskills.interfaces.productInterface.Removable;
+import com.teachmeskills.interfaces.productInterface.Searchable;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+public class ShoppingCartService <T extends ShoppingCart> implements Removable<T>, Searchable<T>, Comparator<Product> {
+
+    public void addProductToShoppingCart(T shoppingCart,
+                                         final Product product,
+                                         final int quantityProduct,
+                                         final Shop shop) throws RuntimeException {
+
+        final ShopService<Shop> shopService = new ShopService<>();
```

Comment:
I think there is a better way to create a shop service object in one place and pass the object to this method than to create a new object each time